### PR TITLE
Moved labels to enable parallelism pipeline in Luke model

### PR DESCRIPTION
**Pull request by sushmanthreddy**
# What does this PR do?

As suggested in the [#22561](https://github.com/huggingface/transformers/issues/22561),moved labels to the same device as logits for the luke model.

@sgugger can u pls review this pr ,there is mistake in the [#22907](https://github.com/huggingface/transformers/pull/22907),I have made necessary changes..

### CODE REVIEW

**Review by amyeroberts — APPROVED** [review]
Thanks for the fix!
